src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Public.Data/PreviewTokenControllerTests.cs

```diff
@@ -0,0 +1,340 @@
+            );
+        }
+
+        [Theory]
+        [MemberData(nameof(NonDraftDataSetVersionStatuses))]
+        public async Task DataSetVersionIsNotDraft_ReturnsValidationProblem(DataSetVersionStatus status)
+        {
+            DataSet dataSet = DataFixture.DefaultDataSet();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion()
+                .WithDataSet(dataSet)
+                .WithStatus(status)
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            var response = await CreatePreviewToken(dataSetVersion.Id, "Label");
+
+            var validationProblem = response.AssertValidationProblem();
+
+            validationProblem.AssertHasError(
+                expectedCode: ValidationMessages.DataSetVersionStatusNotDraft.Code,
+                expectedMessage: ValidationMessages.DataSetVersionStatusNotDraft.Message,
+                expectedPath: nameof(PreviewTokenCreateRequest.DataSetVersionId).ToLowerFirst());
+        }
+
+        [Fact]
+        public async Task DataSetVersionIdIsEmpty_ReturnsValidationProblem()
+        {
+            var response = await CreatePreviewToken(dataSetVersionId: Guid.Empty, "Label");
+
+            var validationProblem = response.AssertValidationProblem();
+
+            Assert.Single(validationProblem.Errors);
+            validationProblem.AssertHasNotEmptyError(nameof(PreviewTokenCreateRequest.DataSetVersionId).ToLowerFirst());
+        }
+
+        [Fact]
+        public async Task LabelIsEmpty_ReturnsValidationProblem()
+        {
+            var response = await CreatePreviewToken(Guid.NewGuid(), label: string.Empty);
+
+            var validationProblem = response.AssertValidationProblem();
+
+            Assert.Single(validationProblem.Errors);
+            validationProblem.AssertHasNotEmptyError(nameof(PreviewTokenCreateRequest.Label).ToLowerFirst());
+        }
+
+        [Fact]
+        public async Task LabelAboveMaximumLength_ReturnsValidationProblem()
+        {
+            var response = await CreatePreviewToken(Guid.NewGuid(), label: new string('A', count: 101));
+
+            var validationProblem = response.AssertValidationProblem();
+
+            Assert.Single(validationProblem.Errors);
+            validationProblem.AssertHasMaximumLengthError(nameof(PreviewTokenCreateRequest.Label).ToLowerFirst(),
+                maxLength: 100);
+        }
+
+        [Fact]
+        public async Task NoDataSetVersion_ReturnsNotFound()
+        {
+            var response = await CreatePreviewToken(dataSetVersionId: Guid.NewGuid(), "Label");
+            response.AssertNotFound();
```

Comment:
Is it worth using the new ValidationUtils.NotFoundResult() here?  Possibly not so useful cos there's only one parameter that could lead to a not found occurring - yeah, forget it!

Reply:
I probably need to revisit that work anyway at some point so that we can do a `SingleOrNotFoundAsync<Guid>(someId, someParameterPath)` which under the hood would use `ValidationUtils.NotFoundResult()`

Reply:
Ah yes, I did try to do this already after I saw that you introduced it but hit the issue where `SingleOrNotFoundAsync` is currently more convenient so I abandoned it for now. It would be good to make the improvement you suggested, thanks. 🙂